src/vdxf/classes/requestobjects/LoginRequestDetails.ts

```diff
@@ -0,0 +1,304 @@
+}
+
+export interface CallbackUriJson {
+  type: number;
+  uri: string;
+}
+export interface LoginPermission {
+  type: BigNumber;
+  identity: CompactIdAddressObject;
+}
+
+export interface CallbackUri {
+  type: BigNumber;
+  uri: string;
+}
+
+export interface LoginRequestDetailsJson {
+  version: number;
+  requestid: string;
+  flags: number;
+  permissions?: Array<LoginPermissionJson>;
+  callbackuri?: Array<CallbackUriJson>;
+  expirytime?: number;
+}
+
+export class LoginRequestDetails implements SerializableEntity {
+  version: BigNumber;
+  flags?: BigNumber;  
+  requestId: string;
+  permissions?: Array<LoginPermission>;
+  callbackUri?: Array<CallbackUri>;
+  expiryTime?: BigNumber;
+
+  // Version
+  static DEFAULT_VERSION = new BN(1, 10)
+  static VERSION_FIRSTVALID = new BN(1, 10)
+  static VERSION_LASTVALID = new BN(1, 10)
+
+  static FLAG_HAS_PERMISSIONS = new BN(1, 10);
+  static FLAG_HAS_CALLBACK_URI = new BN(2, 10);
+  static FLAG_HAS_EXPIRY_TIME = new BN(4, 10);
+
+  // Permission Types
+  static REQUIRED_ID = new BN(1, 10);
+  static REQUIRED_SYSTEM = new BN(2, 10);
+  static REQUIRED_PARENT = new BN(3, 10);
+
+  // Callback URI Types
+  static TYPE_WEBHOOK = new BN(1, 10);
+  static TYPE_REDIRECT = new BN(2, 10);
+  static TYPE_DEEPLINK = new BN(3, 10);
+
+  constructor(
+    request?: LoginRequestDetailsInterface 
+  ) {
+
+    this.version = request?.version || LoginRequestDetails.DEFAULT_VERSION;
+    this.requestId = request?.requestId || '';
+    this.flags = request?.flags || new BN(0, 10);
+    this.permissions = request?.permissions || null;
+    this.callbackUri = request?.callbackUri || null;
+    this.expiryTime = request?.expiryTime || null;
+  }
+
+  hasPermissions(): boolean {   
+      return this.flags.and(LoginRequestDetails.FLAG_HAS_PERMISSIONS).eq(LoginRequestDetails.FLAG_HAS_PERMISSIONS);
+  }
+
+  hasCallbackUri(): boolean {
+    return this.flags.and(LoginRequestDetails.FLAG_HAS_CALLBACK_URI).eq(LoginRequestDetails.FLAG_HAS_CALLBACK_URI);
+  }
+
+  hasExpiryTime(): boolean {
+    return this.flags.and(LoginRequestDetails.FLAG_HAS_EXPIRY_TIME).eq(LoginRequestDetails.FLAG_HAS_EXPIRY_TIME);
+  }
+
+  getByteLength(): number {
+    this.setFlags(); // Ensure flags are set correctly for length calculation
+    let length = 0;
+
+    length += varint.encodingLength(this.flags);
+    length += 20; // requestId hash length
+
+    if (this.hasPermissions()) {
+
+      length += varuint.encodingLength(this.permissions.length);      
+        for (let i = 0; i < this.permissions.length; i++) {
+          length += varint.encodingLength(new BN(this.permissions[i].type));
+          length += this.permissions[i].identity.getByteLength();
+        }      
+    }
+
+    if (this.hasCallbackUri()) {
+      length += varuint.encodingLength(this.callbackUri.length);
+        for (let i = 0; i < this.callbackUri.length; i++) {
+          length += varint.encodingLength(new BN(this.callbackUri[i].type));
```

Comment:
I'll put another varuint/compactSize comment here but lets just say it applies to all values that wouldn't be negative in other files as well unless trying to maintain compat with daemon (times/flags/types etc.)